src/rp2_common/hardware_spi/include/hardware/spi.h

```diff
@@ -197,27 +207,37 @@ static inline void spi_set_slave(spi_inst_t *spi, bool slave) {
  *  \ingroup hardware_spi
  *
  * \param spi SPI instance specifier, either \ref spi0 or \ref spi1
- * \return 0 if no space is available to write. Non-zero if a write is possible
+ * \return false if no space is available to write. True if a write is possible
  *
  * \note Although the controllers each have a 8 deep TX FIFO, the current HW implementation can only return 0 or 1
  * rather than the space available.
  */
-static inline size_t spi_is_writable(spi_inst_t *spi) {
+static inline bool spi_is_writable(const spi_inst_t *spi) {
     // PL022 doesn't expose levels directly, so return values are only 0 or 1
```

Comment:
Should this now say "...so return value is a boolean" ?

Reply:
it should